src/expressions/refExpression.ts

```diff
@@ -216,27 +230,27 @@ export class RefExpression extends Expression {
   public _fillRefSubstitutions(typeContext: DatasetFullType, indexer: Indexer, alterations: Alterations): FullType {
     var myIndex = indexer.index;
     indexer.index++;
-    var nest = this.nest;
-
+    var { nest, ignoreCase, name } = this;
     // Step the parentContext back; once for each generation
     var myTypeContext = typeContext;
     while (nest--) {
       myTypeContext = myTypeContext.parent;
       if (!myTypeContext) throw new Error('went too deep on ' + this.toString());
     }
 
+    var myName = ignoreCase ? findPropertyCI(myTypeContext.datasetType, name) : name;
+
     // Look for the reference in the parent chain
     var nestDiff = 0;
-    while (myTypeContext && !myTypeContext.datasetType[this.name]) {
+    while (myTypeContext && !myTypeContext.datasetType[myName]) {
```

Comment:
there is a very subtle bug here. if `myTypeContext.datasetType` is `{ 'null': 'STRING' }` and `name` is `"blah"` and `ignoreCase` is set. Then `myName  === null` and `myTypeContext.datasetType[myName]` will return a truthy value :-p. The fix is trivial. Bonus points if you can fix this with a test first (TDD)